oak-run-commons/src/main/java/org/apache/jackrabbit/oak/index/indexer/document/DocumentStoreIndexerBase.java

```diff
@@ -207,6 +212,29 @@ private List<FlatFileStore> buildFlatFileStoreList(NodeState checkpointedState,
         return storeList;
     }
 
+    public IndexStore buildIncrementalStore(String initialCheckpoint, String finalCheckpoint) throws IOException, CommitFailedException {
```

Comment:
Can we just have a `buildStore` which returns IndexStore and taken in `Options` which represent what type if indexing to be done

Reply:
On the surface both need to get checkpoints either between 0 -> c1 or c1 -> c2. So, just having it goverened by options should make it simpler to understand. 
We probably don't need to have 2 different interfaces. But need to identify type (FFS, TreeFS), not sure if have to identify it as iincremental.

Reply:
Created overloaded method buildStore